### ISSUE TITLE
Unexpected behavior with `np.nan` only initialized vectors

### ISSUE THREAD

**Issue by a-monsch** [scikit-hep/vector#138]
I am currently working with this package in combination with pandas on an assignment for study exercises in particle physics and have encountered an unexpected behavior. For a homogenized structure of my data  I use `np.nan` to fill some missing values (particles). Consequently, it often comes to a situation where an initialization  of a vector obj/array/... with `np.nan` occurs. This works perfectly fine, especially the property query, except for `pseudorapidity`, `rapidity` and other related quantities.

A small example:

```
import vector
import numpy as np

# initially, this part is stored inside a pandas DataFrame
v = np.array([[4.0, 3.0, 2.0, 1.0], [np.nan, np.nan, np.nan, np.nan]]).view(
                        [("E", float), ("px", float), ("py", float), ("pz", float)]).view(
                        vector.MomentumNumpy4D)


# this creates a [True, False] mask; as expected 
mask1 = np.abs(v.rapidity) < 2.5   

# This creates a [True, True] mask, since the second value is not np.nan but 0.0
mask2 = np.abs(v.pseudorapidity) < 2.5  
```
My expectation would be that, in this case, any quantity of an object that is initialized with (only) `np.nan` should also return a `np.nan` and not an actual number.

Is this behavior, especially for a case described above, intended?

P.S.: I know that via `vector.awk` there is a possibility to work with inhomogeneous data structures, but in our workgroup we decided against that and for an approach using pandas.

**Comment by a-monsch** [comment]
Thank you very much for the detailed and fast answer. 

According to #36, you always compare with ROOT. In my first intention, this was not even the point, since it makes no sense to calculate anything at all from a vector that only contains `np.nan`, but nice to see that ROOT actually does it as expected, although only taking into account the `z` and `phi` components as you pointed out. 

The solution with the usage of `lib.absolute(lib.sign(z))` would fix the problem. Will this be implemented in the same or similar form in one of the next versions, or will it take longer, since the TODO list at #36 for the 4D vectors has not been addressed yet as far as I can see?

**Comment by jpivarski** [comment]
I'm implementing the `lib.absolute(lib.sign(z))` solution now. PR #36 is waylaid because the author had to work on other things. We won't get complete parity with ROOT until those tests are added, but we can fix this one that you noticed now: it will be implemented in PR #139.
